packages/engine/lib/stateful/src/field/mod.rs

```diff
@@ -1,21 +1,14 @@
-//! Provides a memory format similar to Arrows `Field`, `DataType`, and `Schema`.
+//! Provides datastructures to organize field specifications and data types.
 //!
-//! As only a selection of Arrow [`DataType`]s is supported at hEngine and additional constraints on
-//! the Arrow column names are required to distinguish the different [`FieldSource`]s and
-//! [`FieldScope`]s, this module abstracts over Arrow types. It defines [`FieldType`], which uses
-//! Arrow types underneath and adds an `Any` type, which can represent arbitrary data and
-//! [`FieldSpec`] to associate a name with the [`FieldType`].
+//! Each supported data type is represented by a [`FieldType`]. It's associated with a name in
+//! [`FieldSpec`]. Depending on the underlying [`FieldTypeVariant`], types may be nested.
 //!
-//! In order to use Arrows [`RecordBatch`] to use its memory format internally, an Arrow [`Schema`]
-//! can be created from a [`FieldSpecMap`], which maps a [`RootFieldKey`]s to [`RootFieldSpec`]s. A
-//! [`RootFieldSpec`] associates a [`FieldSpec`] with a [`FieldScope`] and a [`FieldSource`] and can
-//! only nest [`FieldSpec`]s. The [`RootFieldKey`] contains the name of the corresponding field and
-//! encodes the [`FieldScope`] and a [`FieldSource`] to be used for looking up Arrow columns.
+//! In order to store a [`FieldSpec`] into memory, a unique name is required, which is encapsulated
+//! in [`RootFieldKey`]. This is mapped to a [`RootFieldSpec`] inside a [`FieldSpecMap`]. A
+//! [`FieldSpecMap`] can be converted to an Arrow [`Schema`] to be stored in memory.
```

Comment:
Maybe "Top-level Fields (i.e. non-nested ones) have additional specification, defined in a [`RootFieldSpec`]. These have associated unique identifiers, [`RootFieldKey`]s, which are mapped against their [`RootFieldSpec`] in a [`FieldSpecMap`]. The [`FieldSpecMap`] is then used to generate an Arrow [`Schema`] to outline how the data is stored in memory"

(By the way, we can consider renaming `FieldSpecMap` to `RootFieldSpecMap` if we want, but it is quite a mouthful)
(Also lets avoid referring to `RootFieldKey`s as "names" because that's a bit confusing)